### ISSUE TITLE
Blazor WebAssembly dotnet watch run --pathbase=/ doesn't work in .NET 8

### ISSUE THREAD

**Issue by MaciekWin3** [dotnet/aspnetcore#52294]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Hi, after the latest update to .NET 8, I discovered that the `--pathbase` flag is not working as it should. In the previous version, the app was served from the specified path, but in the new version, it is being ignored. I am using this option to debug my Blazor Extension in Azure DevOps as I need to specify a different path than the default one. The command I am using is:
```
dotnet watch run --pathbase=/dist
```
Then the app is served from /dist folder, and I can access my app and debug easily using hot reloading. At the root level, I get this message: "The server is configured only to handle request URIs within the PathBase '/dist'." On .NET 8, this flag is simply ignored, and the app is served from the default path.

### Expected Behavior

I expected the app to run from specified path with the `dotnet watch run --pathbase=path` command after updating to .NET 8, like it did before. But now, the flag isn't working, and the app runs from the default path instead. 

### Steps To Reproduce

To reproduce the issue, launch a Blazor WebAssembly project using the command dotnet watch run --pathbase=/dist and observe whether the application is served at the specified path ('/dist' in this case).

### .NET Version

8.0.100

**Comment by mkArtakMSFT** [comment]
@tmat is this something you've potentially changed? I don't recall us making any changes in this area during .NET 8.

**Comment by afouadr** [comment]
Working:
dotnet run --pathbase=<path>

**Comment by tmat** [comment]
Hot Reload is disabled when running the above command line with SDK 7:

```
>dotnet watch --verbose run --pathbase=/dist
dotnet watch ⌚ Did not find a HotReloadProfile or running a non-default command. Watching with legacy behavior.
```

We no longer disable it with SDK 8.

A workaround is to explicitly disable Hot Reload `--no-hot-reload` like so:

```
dotnet watch run --no-hot-reload --pathbase=/dist
```

**Comment by KenDRhyD** [comment]
I ran into this recently. I have .Net 7 Blazor app that is working on my MacOS system (via VSCode), and I attempted to update to .Net 8 by installing the 8.0.2 SDK and then changing the csproj file to update the framework and WebAssembly versions, and I was getting the same error message.

So I deleted the bin and obj folders completely and tried again and  found that a "dotnet run" allows the "https://localhost:7035/" URL to invoke my code, but a "dotnet watch" reports the "The server is configured only to handle request URIs within the PathBase '/SMaLL'." error! And is I append "/SMaLL" to the URL then all resources report 404 error!

At least I can use the run command, but it is a pain to have to kill and restart with each code change, as I was able to use watch before.
